src/tasks/memory.py

```diff
@@ -68,3 +77,463 @@ def _run(self, exp_win, ctl_win):
+Remember the locations of the number pairs on screen.
+You will be asked to recall them sequentially, and win one point per correct pair.
+If you can remember all the pairs, you will win the bonus points.
+"""
+
+    def __init__(self, items_list, total_possible_points, *args, **kwargs):
+        super().__init__(**kwargs)
+        self.total_possible_points = total_possible_points
+        self.item_list = pd.read_csv(items_list, sep="\t", index_col=0).to_dict("records")
+        self.duration = len(self.item_list)  # not sure if this is needed
+        self._progress_bar_refresh_rate = 2
+        self.trials = data.TrialHandler(self.item_list, 1, method="sequential")
+        self.grid_size = (6, 4)
+        # self.direction_keys = {"left": "a", "right": "d", "up": "w", "down": "s"}  # laptop
+        # self.direction_values = {"a": DIRECTIONS["left"],
+        #                         "d": DIRECTIONS["right"],
+        #                         "w": DIRECTIONS["up"],
+        #                         "s": DIRECTIONS["down"]}  # laptop
+        # self.confidence_keys = {'yes': 'k', 'no': 'l'}  # laptop
+        # self.rating_pygame = {"left": key.A, "right": key.D}  # laptop
+        self.confidence_keys = {'yes': 'a', 'no': 'b'}  # mri
+        self.direction_keys = {"left": "l", "right": "r", "up": "u", "down": "d"}   # mri
+        self.direction_values = {"l": DIRECTIONS["left"],
+                                 "r": DIRECTIONS["right"],
+                                 "u": DIRECTIONS["up"],
+                                 "d": DIRECTIONS["down"]}  # mri
+        self.rating_pygame = {"left": key.L, "right": key.R}  # mri
+
+    def _instructions(self, exp_win, ctl_win):
+        screen_text = visual.TextStim(
+            exp_win,
+            text=self.instruction,
+            alignText="center",
+            color="white",
+            wrapWidth=config.WRAP_WIDTH,
+        )
+        instruction_duration = 10
+        for _ in range(config.FRAME_RATE * instruction_duration):
+            screen_text.draw(exp_win)
+            if ctl_win:
+                screen_text.draw(ctl_win)
+            yield True
+
+        message = (
+            f"In this session, "
+            f"you can earn up to {self.total_possible_points} points.")
+        duration = 3
+        screen_text.text = message
+        for _ in range(config.FRAME_RATE * duration):
+            screen_text.draw(exp_win)
+            if ctl_win:
+                screen_text.draw(exp_win)
+            yield True
+
+    def _setup(self, exp_win):
+        # make pares of rectangels and text stim
+        self.grid = [(
+            visual.TextStim(
+                exp_win, text="", pos=(-0.8 + x * 0.25, 0.7 - y * 0.25),
+                alignText="center", color="white"
+            ),
+            visual.Rect(
+                exp_win, size=(0.2, 0.2), pos=(-0.8 + x * 0.25, 0.7 - y * 0.25),
+                fillColor=None,
+            )
```

Comment:
Grid was not centered on my screen, tried to adapt size of grid element and centering so that it adapts to grid size (if ever).

```suggestion
            visual.TextStim(
                exp_win, text="",
                pos=( (x-(self.grid_size[0]-1)/2.) * box_size,
                    .8 - y * box_size),
                alignText="center", color="white"
            ),
            visual.Rect(
                exp_win, size=(box_size*.8, box_size*.8),
                pos=( (x-(self.grid_size[0]-1)/2.) * box_size,
                    .8 - y * box_size),
                fillColor=None,
            )
```

Reply:
Fantastic - I definitely paid zero effort into aligning the grids